homeassistant/components/device_tracker/google_maps.py

```diff
@@ -0,0 +1,83 @@
+"""
+Support for Google Maps location sharing.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.google_maps/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, SOURCE_TYPE_GPS)
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['locationsharinglib==0.4.0']
+
+CREDENTIALS_FILE = '.google_maps_location_sharing.cookies'
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_scanner(hass, config: ConfigType, see, discovery_info=None):
+    """Set up the scanner."""
+    scanner = GoogleMapsScanner(hass, config, see)
+    return scanner if scanner.success_init else None
```

Comment:
`setup_scanner` doesn't want the scanner instance, it just wants a boolean, where true means successful platform setup.

This will work, but it's less clear.

Reply:
A lot of `device_tracker` platforms seem to do __exactly__ this. I don't think this is blocking the merge of this component? Anyway, I will make the change...

```
homeassistant/components/device_tracker/actiontec.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/aruba.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/asuswrt.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/bbox.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/bt_home_hub_5.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/cisco_ios.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/fritz.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/google_maps.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/hitron_coda.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/keenetic_ndms2.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/luci.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/mikrotik.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/netgear.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/nmap_tracker.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/sky_hub.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/snmp.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/swisscom.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/tado.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/thomson.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/ubus.py:    return scanner if scanner.success_init else None
homeassistant/components/device_tracker/xiaomi.py:    return scanner if scanner.success_init else None
```

Reply:
`get_scanner` should return the scanner instance but not `setup_scanner`. Platforms that use that latter should not behave like the former.